beets/util/__init__.py

```diff
@@ -442,17 +450,36 @@ def samefile(p1, p2):
     return shutil._samefile(syspath(p1), syspath(p2))
 
 
-def remove(path, soft=True):
-    """Remove the file. If `soft`, then no error will be raised if the
-    file does not exist.
+def remove(path, soft=True, use_trash=True):
+    """Remove the file specified at `path`, using `trash` executable by default
+    (if found) or via the command specified in the `remove_command` setting if
+    found on $PATH. Otherwise, delete the file immediately. If `soft`, then no
+    error will be raised if the file does not exist.
     """
     path = syspath(path)
     if soft and not os.path.exists(path):
         return
-    try:
-        os.remove(path)
-    except (OSError, IOError) as exc:
-        raise FilesystemError(exc, 'delete', (path,), traceback.format_exc())
+    # Use `remove_command` setting, if available. Otherwise assume `trash`.
+    if config["remove_command"]:
+        remove_command = config["remove_command"].as_str()
+    else:
+        remove_command = "trash"
+    # Use trash if `remove_command` is found on $PATH, else delete immediately.
+    # Change which() to shutil.which() once Python <3.3 support is dropped
+    if use_trash and which(remove_command) is not None:
```

Comment:
```suggestion
    # Use trash if `remove_command` (which defaults to `trash`) is found on $PATH,
    # else delete immediately.
    # Change which() to shutil.which() once Python <3.3 support is dropped
    remove_command = config["remove_command"].as_str()
    if use_trash and which(remove_command) is not None:
```

I think with confuse, you can immediately access the value: The `config_default.yaml` file will always be loaded, so this will return the default `trash` if the user didn't specify anything else. Accessing the configuration like this is a common pattern in beets.

Reply:
Agreed. I removed the extraneous fallback in the latest commit. Aside from any areas where you think opting out from removal to trash is needed, I believe we are all good to go here.